src/families/crypto_org/js-synchronisation.js

```diff
@@ -0,0 +1,55 @@
+//@flow
+import type { Account } from "../../types";
+import type { GetAccountShape } from "../../bridge/jsHelpers";
+import { makeSync, makeScanAccounts, mergeOps } from "../../bridge/jsHelpers";
+
+import { getAccount, getOperations } from "./api";
+import { TESTNET_CURRENCY_ID } from "./logic";
+
+const getAccountShape: GetAccountShape = async (info) => {
+  const { id, address, initialAccount, currency } = info;
+  const oldOperations = initialAccount?.operations || [];
+
+  const useTestNet = currency.id == TESTNET_CURRENCY_ID ? true : false;
+  // get the current account balance state depending your api implementation
+  const {
+    blockHeight,
+    balance,
+    bondedBalance,
+    redelegatingBalance,
+    unbondingBalance,
+    commissions,
+  } = await getAccount(address, useTestNet);
+
+  // Merge new operations with the previously synced ones
+  let startAt = 0;
+  let maxIteration = 20;
+  let operations = oldOperations;
+  let newOperations = await getOperations(id, address, startAt++, useTestNet);
+  do {
+    operations = mergeOps(operations, newOperations);
+    newOperations = await getOperations(id, address, startAt++, useTestNet);
+  } while (--maxIteration && newOperations.length != 0);
```

Comment:
This pagination handling logic should be in the api i think.
I think mergeOps is quite a resource consuming operation (merges arrays by comparing ids).

Plus, it always starts at zero here, and does not take into consideration the previous synced operation.
LedgerLive sync is iterative, every 2 minutes, each account is synced, but inherits from its previously fetched state.

Here, i don't see that this is taken into consideration, and might burden the api heavily. Can't you set a minimum blockHeight as a starting point when fetching operations ?

Reply:
unfortunately api does not take block height but index. If I can store this index somewhere in Operation, then I can optimize this loop